en/services/parameter_ext.md

```diff
@@ -0,0 +1,238 @@
+
+
+## Parameter Encoding {#parameter_encoding}
+
+Parameters names/ids are set in the `param_id` field of messages where they are used.
+The `param_id` string can store up to 16 characters. 
+The string is terminated with a NULL (`\0`) character if there are less than 16 human-readable chars, and without a null termination byte if the length is exactly 16 chars.
+
+> **Note** Names (as above) are the same as for the [Parameter Protocol](../services/parameter.md#parameter_encoding).
+  Values (below) are encoded differently.
+
+Values are byte-wise encoded *within* the `param_value` field, which is a `char[128]`. <!-- how are they encoded? -->
+The `param_type` ([MAV_PARAM_EXT_TYPE](#MAV_PARAM_EXT_TYPE)) is used to indicate the actual type of the data so that it can be decoded by the recipient.
+Supported types are: 8, 16, 32 and 64-bit signed and unsigned integers, 32 and 64-bit floating point numbers, and a "custom type" which may used for strings. <!-- why custom? shouldit be string, and then new types added for non string usage of the same field -->
+
+<!--  **Note** A byte-wise conversion is needed, rather than a simple cast, to enable larger integers to be exchanged (e.g. 1E7 scaled integers can be useful for encoding some types of data, but lose precision if cast directly to floats).
+-->
+
+
+## Parameter Caching {#parameter_caching}
+
+A GCS or other component may choose to maintain a cache of parameter values for connected components/systems, in order to reduce the time required to display values and reduce MAVLink traffic.
+
+The cache can be populated initially by first [reading the full parameter list](#read_all) at least once, and then updated by monitoring for [PARAM_EXT_ACK](#PARAM_EXT_ACK) messages with `PARAM_ACK_ACCEPTED` (which are emitted whenever a parameter is successfully [written/changed](#write)).
+
+A system may also monitor for [PARAM_EXT_VALUE](#PARAM_EXT_VALUE) originating from other components/systems requesting parameter values. 
+
+> **Note** Cache synchronisation is not guaranteed; a component may [miss parameter update messages](#monitoring_unreliable) due to changes by other components.
+
+
+<!--
+## Multi-System and Multi-Component Support
+
+MAVLink supports multiple systems in parallel on the same link, and multiple MAVLink enabled components within a system.
+
+Requests to get and set parameters can be sent to individual systems or components. 
+To get a complete parameter list from a system, send the request parameter message with `target_component` set to [MAV_COMP_ID_ALL](../messages/common.md#MAV_COMP_ID_ALL).
+
+All components must respond to parameter request messages addressed to their ID or the ID `MAV_COMP_ID_ALL`. 
+
+> **Tip** *QGroundControl* by default queries all components of the currently connected system (it sends ID `MAV_COMP_ID_ALL`).
+-->
+
+
+<!--
+### Mavgen C API
```

Comment:
There were explicit helpers for working with parameters in the original API, which was documented here (e.g. making it easier to do bytewise copy of types into the messages).  I have commented out, and assume nothing similar available/needed for this protocol?